Sources/XiEditor/LineCache.swift

```diff
@@ -110,13 +131,38 @@ fileprivate class LineCacheState<T>: UnfairLock {
     /// Updates the state by applying a delta. The update format is detailed in the
     /// [xi-core docs](http://xi-editor.github.io/xi-editor/docs/frontend-protocol.html#view-update-protocol).
     func applyUpdate(update: [String: AnyObject]) -> InvalSet {
+        // parse annotations
+        let annotationData = (update["annotations"] as! [[String: AnyObject]])
+
+        for annotationType in AnnotationType.allCases {
```

Comment:
I didn't think it through, but it seems like iterating over `annotationData` could create a simpler solution here 🤔

Reply:
I also think we can totally move this to a helper function, `updateAnnotations`. `applyUpdate` is already over a hundred lines long, and this new code is pretty independent.